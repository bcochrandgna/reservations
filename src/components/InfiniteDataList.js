// @flow
import React from 'react';
import { FlatList, Text } from 'react-native';
import { Reservation } from '../types/Reservation';

type Props = {
  data: Array<Reservation>,
  fetchMore: (options: any) => void,
  refetch: () => void,
  networkStatus: number,
  renderItem: (item: Reservation) => any,
};

const InfiniteDataList = (props: Props) => {
  const {
    data, fetchMore, refetch, networkStatus, renderItem,
  } = props;

  if (!data || data.length === 0) {
    return <Text>No Reservations.</Text>;
  }
  return (
    <FlatList
      data={data}
      refreshing={networkStatus === 4}
      onRefresh={() => refetch()}
      onEndReachedThreshold={0.5}
      onEndReached={fetchMore}
      renderItem={({ item }) => renderItem(item)}
      keyExtractor={item => item.id}
    />
  );
};

export default InfiniteDataList;
