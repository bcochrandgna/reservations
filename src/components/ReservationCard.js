// @flow
import React from 'react';
import { View, Text } from 'react-native';
import moment from 'moment';
import styles from '../styles/components/ReservationCard';
import i18n from '../helpers/i18n';
import LabelValue from './LabelValue';
import { Reservation } from '../types/Reservation';

type Props = {
  reservation: Reservation,
};

const ReservationCard = (props: Props) => {
  const { reservation } = props;

  if (!reservation || !reservation.name) return null;

  const formattedArrivalDate = reservation.arrivalDate
    && moment(new Date(reservation.arrivalDate)).format('dddd, MMMM Do YYYY');
  const formattedDepartureDate = reservation.departureDate
    && moment(new Date(reservation.departureDate)).format('dddd, MMMM Do YYYY');

  return (
    <View style={styles.container}>
      <View style={styles.titleContainer}>
        <Text style={styles.titleText}>{reservation.name}</Text>
        <Text>{reservation.hotelName}</Text>
      </View>
      <View style={styles.bottomContainer}>
        <View>
          <LabelValue label={i18n.t('arrival_date')} value={formattedArrivalDate} />
          <LabelValue label={i18n.t('departure_date')} value={formattedDepartureDate} />
        </View>
        <View />
      </View>
    </View>
  );
};

export default ReservationCard;
