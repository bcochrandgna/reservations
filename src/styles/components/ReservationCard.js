import { StyleSheet } from 'react-native';
import {
  borderColor,
  normalTextColor,
  mediumTextSize,
  titleTextColor,
  cardBackgroundColor,
  standardMargin,
  standardPadding,
  cardShadedColor,
} from '..';

export default StyleSheet.create({
  container: {
    borderWidth: 1,
    borderColor,
    marginHorizontal: standardMargin,
    marginBottom: standardMargin,
    backgroundColor: cardBackgroundColor,
  },
  dateText: {
    color: normalTextColor,
    marginLeft: standardMargin,
  },
  titleText: {
    fontSize: mediumTextSize,
    color: titleTextColor,
  },
  titleContainer: {
    padding: standardPadding,
  },
  bottomContainer: {
    marginTop: standardMargin,
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: cardShadedColor,
    padding: standardPadding,
  },
});
