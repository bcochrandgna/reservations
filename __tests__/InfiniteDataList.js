import React from 'react';
import { shallow } from 'enzyme';
import InfiniteDataList from '../src/components/InfiniteDataList';
import reservations from '../fixtures/reservations';
import ReservationCard from '../src/components/ReservationCard';

const renderItem = item => <ReservationCard reservation={item} />;

describe('InfiniteDataList', () => {
  it('renders correctly with null props', () => {
    const wrapper = shallow(<InfiniteDataList />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with props', () => {
    const wrapper = shallow(
      <InfiniteDataList data={reservations} networkStatus={1} renderItem={renderItem} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
