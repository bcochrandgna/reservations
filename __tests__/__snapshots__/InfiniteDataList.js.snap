// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfiniteDataList renders correctly with null props 1`] = `
<Text>
  No Reservations.
</Text>
`;

exports[`InfiniteDataList renders correctly with props 1`] = `
<FlatList
  data={
    Array [
      Object {
        "arrivalDate": "03/01/2019",
        "departureDate": "03/03/2019",
        "hotelName": "Test Hotel 1",
        "id": "1",
        "name": "Test Name 1",
      },
      Object {
        "arrivalDate": "03/01/2019",
        "departureDate": "03/03/2019",
        "hotelName": "Test Hotel 2",
        "id": "2",
        "name": "Test Name 2",
      },
      Object {
        "arrivalDate": "03/01/2019",
        "departureDate": "03/03/2019",
        "hotelName": "Test Hotel 3",
        "id": "3",
        "name": "Test Name 3",
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={0.5}
  onRefresh={[Function]}
  refreshing={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
